### ISSUE TITLE
tf.nn.conv2d_transpose?

### ISSUE THREAD

**Issue by engineeringmath** [jakeret/tf_unet#24]
Hi,

Thank you for the great work!

Reading the original U-Net paper and comparing with your implementation, it looks like you have used tf.nn.conv2d_transpose to implement "up-sampling followed by convolution". But the tensorflow documentation says that this function "the transpose (gradient) of conv2d". I am wondering if this is the right operation to use here.

**Comment by jakeret** [comment]
Interesting point. Are you aware of an alternative?
I have seen other projects using this function for the up-sampling as well

**Comment by engineeringmath** [comment]
thanks for the response.

Not sure. But if we are supposed to do up-sampling followed by convolution, can we use tf.image.resize_images (with nearest-neighbor option) to up-sample and then apply a regular conv2d?

Thanks again

**Comment by jakeret** [comment]
I need to check. If I remember correctly, I was using a similar (image) function somewhere else and it was causing issues with the gradients
